src/odemis/acq/feature.py

```diff
@@ -119,3 +143,322 @@ def load_project_data(path: str) -> dict:
+        positions = []
+        for f in self.features:
+            if f.status.value == FEATURE_DEACTIVE:
+                continue
+            positions.append(f.pos.value)
+
+        stage_movement_time = 0
+        for start, end in zip(positions[0:-1], positions[1:]):
+            stage_movement_time += estimate_stage_movement_time(
+                                    stage=self.stage,
+                                    start_pos={"x": start[0], "y": start[1]},
+                                    end_pos={"x": end[0], "y": end[1]},
+                                    axes=["x", "y"], independent_axes=True)
+
+        # add the time to wait for the stage to settle
+        expected_stage_time = stage_movement_time + STAGE_WAIT_TIME * len(positions)
+        logging.debug(f"Estimated total stage movement time {stage_movement_time}s")
+
+        # estimate the time to move the objective
+        expected_objective_time = 1 * len(positions) # 1 second for objective movement (estimated)
+        logging.debug(f"Estimated total objective movement time {expected_objective_time}s")
+
+        # total movement time
+        expected_movement_time = expected_stage_time + expected_objective_time
+        logging.debug(f"Estimated total movement time {expected_movement_time}s")
+        return expected_movement_time
+
+    def estimate_total_time(self) -> float:
+        """Estimate the total time for the acquisition task."""
+        n_active_features = sum(f.status.value != FEATURE_DEACTIVE for f in self.features)
+        expected_movement_time = self.estimate_movement_time()
+        expected_acq_time = self.estimate_acquisition_time()
+        total_time = expected_acq_time * n_active_features + expected_movement_time
+        logging.debug(f"Estimated total time {total_time}s for {n_active_features} features")
+        return total_time
+
+    def run(self) -> Tuple[List[model.DataArray], Exception]:
+        """Run the acquisition task."""
+        exp = None
+        self._future._task_state = RUNNING
+        self._future.set_progress(end=time.time() + self.estimate_total_time() + 2)
+        try:
+            with self._future._task_lock:
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                # move to FM IMAGING posture
+                current_posture = self.pm.getCurrentPostureLabel()
+                logging.debug(f"Current Posture: {POSITION_NAMES[current_posture]}")
+
+                if current_posture != FM_IMAGING:
+                    raise ValueError(f"Currently only supported when at {POSITION_NAMES[FM_IMAGING]} posture")
+                    # TODO: enable this when the cryo switch is fixed
+                    logging.debug(
+                        f"Moving to {POSITION_NAMES[FM_IMAGING]} from {POSITION_NAMES[current_posture]}"
+                    )
+                    f = self.pm.cryoSwitchSamplePosition(FM_IMAGING)
+                    f.result()
```

Comment:
In order to be able to cancel the task while it's switching posture, you should store the Future as the "running_subf". Like that:
```suggestion
                    self._future.running_subf = self.pm.cryoSwitchSamplePosition(FM_IMAGING)
                    self._future.running_subf.result()
```